src/addons/dualdirectional.cpp

```diff
@@ -157,7 +157,7 @@ void DualDirectionalInput::process()
             // this also avoids accidentally masking gamepad inputs with the lack of dual inputs
             if (gamepad->options.dpadMode == options.dualDirDpadMode) {
```

Comment:
I've done some tests and this seems to be the logic that gets both regular and DDI working together when both input methods are same.

```c
if (gamepad->options.dpadMode == options.dualDirDpadMode) {
    uint8_t gamepadDpad = gpadToBinary(gamepad->options.dpadMode, gamepad->state);
    if ( socdMode == SOCD_MODE_NEUTRAL ) {
        dualOut = SOCDCombine(socdMode, gamepadDpad);
    } else if ( socdMode != SOCD_MODE_BYPASS ) {
        dualOut = SOCDGamepadClean(dualOut | gamepadDpad, socdMode == SOCD_MODE_SECOND_INPUT_PRIORITY);
    } else {
        dualOut |= gamepadDpad;
    }
}```

I think line 158 to 161 will be replaced by this change.